### PR TITLE
Odyssey Stats: use WPCOM site info as defaults

### LINKED ISSUE #81538: Enable Commercial upgrade notice from Odyssey Stats
`is_commercial` is not available to Jetpack yet, we'll need to add support in Jetpack.

### PR DESCRIPTION
**Pull request by kangzj**
This is not the primary purpose, but it fixes #81538 as well. 

## Proposed Changes

[is_commercial is a WPCOM option](https://github.com/Automattic/jetpack/pull/32595), which we need to query  from WPCOM.As some of the information might not be accurate, we are going to use them as defaults with the options in initial state from Jetpack as source of truth.

The PR proposes to add the site info querying in the process of `siteSelection` middleware in `apps/odyssey-stats/src/routes.ts`, where queries `jetpack/v4/site`, decodes the data, and dispatches Redux actions to bridge the whole data flow.

As the old reducer would override the whole site info with data from WPCOM, the PR has to add a new reducer for action `ODYSSEY_SITE_RECEIVE` which uses site info from wpcom as defaults and overrides it with values from initial state.

## Testing Instructions

* Build Jetpack if needed `jetpack build plugins/jetpack`
* Add a line at: 
https://github.com/Automattic/wp-calypso/blob/f3bf9699554234847c856172c7dfe1511fbc1ab6/apps/odyssey-stats/src/app.tsx#L59
```
setInterval(()=> console.log(store.getState()), 5000);// Or anyone can remind me the  middleware to attach the redux state to window? couldn't find it anywhere anymore 🤷
```
* Build Odyssey Stats `cd apps/odyssey-stats && STATS_PACKAGE_PATH=~/path/to/jetpack/projects/packages/stats-admin yarn dev`
* Open `/wp-admin/admin.php?page=stats#!/stats/day/:siteSlug`
* Check console
* Ensure you could find `is_commercial` in sites -> items -> :blog_id -> options

**Note: the buttons to purchase Stats do not work at the moment and we'll fix those in follow up PRs.**

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/state/sites/reducer.js [hunk 1]
```diff
@@ -85,6 +91,38 @@ export const items = withSchemaValidation( sitesSchema, ( state = null, action )
 			);
 		}
 
+		case ODYSSEY_SITE_RECEIVE: {
+			// Treat the site info from WPCOM as default values for the site, and the info from Odyssey as the source of truth.
+			// This is because the site info from WPCOM is more complete, but the info from Odyssey is more up-to-date.
+			// For example, `options.is_commercial` is not present in the Odyssey site info, but is a remote option value stored in WPCOM.
+			const sites = [ action.site ];
+			const initialNextState = SITES_RECEIVE === action.type ? {} : state;
```
Comment (dognose24) [inline client/state/sites/reducer.js]: Is the condition possible to be true in this `ODYSSEY_SITE_RECEIVE` statement?

File: client/state/sites/reducer.js [hunk 2]
```diff
@@ -85,6 +91,38 @@ export const items = withSchemaValidation( sitesSchema, ( state = null, action )
 			);
 		}
 
+		case ODYSSEY_SITE_RECEIVE: {
+			// Treat the site info from WPCOM as default values for the site, and the info from Odyssey as the source of truth.
+			// This is because the site info from WPCOM is more complete, but the info from Odyssey is more up-to-date.
```
Comment (dognose24) [inline client/state/sites/reducer.js:96]: 👍🏼

**Review by dognose24 — APPROVED** [review]
LGTM! 👍🏼

File: client/state/sites/reducer.js (hunk 1 above)
Comment (kangzj) [reply client/state/sites/reducer.js]: Nice catch! Fixed in 6df921bd81

**Pull request merged**